src/index.js

```diff
@@ -63,14 +84,37 @@ class Vault {
     const hmac = this._hmac(hmacKey, ciphertext)
     const hex = [ salt, hmac, ciphertext ].map(buf => buf.toString('hex')).join('\n')
 
+    return this._pack(id, hex)
+  }
+
+  _decypher (unpacked, derivedKey) {
+    const { hmac, ciphertext } = unpacked
+    const { key, hmacKey, iv } = derivedKey
+    const hmacComp = this._hmac(hmacKey, ciphertext)
+
+    if (Buffer.compare(hmacComp, hmac) !== 0) throw new Error('Integrity check failed')
+
+    const cipherF = crypto.createDecipheriv(CIPHER, key, iv)
+    const buffer = pkcs7.unpad(Buffer.concat([
+      cipherF.update(ciphertext),
+      cipherF.final()
+    ]), 16)
+
+    return buffer.toString()
+  }
+
+  _pack (id, hex) {
     const header = id
       ? `${HEADER};1.2;${AES256};${id}\n`
       : `${HEADER};1.1;${AES256}\n`
 
     return header + hexlify(hex).match(/.{1,80}/g).join('\n')
   }
 
-  async decrypt (vault, id) {
+  _unpack (vault, id) {
+    if (!id) {
+      console.log('no id defined')
+    }
```

Comment:
can you please remove this? ids are used to tag ansible vaults in v1.2 
v1.1 does not use this concept.

Reply:
Thanks :/ Sorry about that. Done.